packages/playwright/src/reporter.ts

```diff
@@ -340,22 +350,37 @@ export default class ReplayPlaywrightReporter implements Reporter {
   async onEnd() {
     try {
       await this.reporter.onEnd();
-      if (!this._foundReplayBrowser) {
-        mixpanelAPI.trackEvent("playwright.warning.reporter-used-without-browser");
 
-        const output: string[] = [];
+      const output: string[] = [];
+
+      const executedProjectWithReplay = !Object.keys(this._projects).some(projectName => {
+        const { executed, usingReplay } = this._projects[projectName];
+        return executed && usingReplay;
+      });
+
+      if (!executedProjectWithReplay) {
+        mixpanelAPI.trackEvent("playwright.warning.reporter-used-without-replay-project");
         output.push(emphasize("None of the configured projects ran using Replay Chromium."));
-        if (!existsSync(getRuntimePath())) {
+      }
+
+      if (!existsSync(getRuntimePath())) {
+        mixpanelAPI.trackEvent("playwright.warning.reporter-used-without-replay-browser");
```

Comment:
I think... putting this warning here doesn't make sense, does it? I was planning to follow up on this in PRO-751 but...wouldn't we error much earlier on when we tried to run the first test but there was no browser to run it with?

Reply:
> wouldn't we error much earlier on when we tried to run the first test but there was no browser to run it with?

Not quite. We can't check what projects are actually executed in the constructor. So we start tracking this information in `onBegin` but we also can't check what projects are actually executed there... so we only gain that knowledge in `onEnd` as at that time we know about all executed tests and related projects.

You are right that all related tests would just fail in such a scenario. It would be a Playwright-generated error though, smth like:
```
  1) [replay-chromium] › test/example.spec.ts:3:5 › basic test ─────────────────────────────────────

    Error: browserType.launch: Failed to launch chromium because executable doesn't exist at /Users/mateuszburzynskiprev/.replay/runtimes/Replay-Chromium.app/Contents/MacOS/Chromium
```

And when that gets logged we can't add a mixpanel event.

Reply:
Okay. I was thinking we could perhaps detect and warn earlier (in `onTestBegin`) but maybe it's better to print the error at the end anyway.

That being said, this check doesn't look right because it's warning _unconditionally_ (even if Replay project wasn't used)

Reply:
> That being said, this check doesn't look right because it's warning unconditionally (even if Replay project wasn't used)

Hm, yee - you are right. It makes sense to only warn about this on runs when Replay was meant to be executed.